### PR TITLE
fix(ThemeContextProvider): Support SSR + Strict Mode

### PR DESCRIPTION
**Pull request by pladaria**

### CODE REVIEW

File: flow-defs/theme-context-provider.js.flow [hunk 1]
```diff
@@ -5,6 +5,17 @@ import type { ThemeConfig } from "./theme";
 declare export var useIsOsDarkModeEnabled: () => boolean;
 declare type Props = {
   theme: ThemeConfig,
+
+  /**
+   * You should use this prop if you use Strict Mode and Server Side Rendering together.
+   * This identifier will be used to generate unique class names for each instance of ThemeContextProvider.
+   * If no identifier is provided, this will fallback to an aunto-incremented id, which will cause
```
Comment (sergivillar) [inline flow-defs/theme-context-provider.js.flow:12]: typo `auto`

**Review by sergivillar — APPROVED** [review]

File: src/theme-context-provider.tsx [hunk 2]
```diff
@@ -70,16 +80,21 @@ const useDefaultHrefDecorator = () => {
...
 
-const ThemeContextProvider: React.FC<Props> = ({theme, children}) => {
+const ThemeContextProvider: React.FC<Props> = ({theme, children, providerId}) => {
+    const [instanceId] = React.useState(() => {
+        if (providerId) {
+            return providerId;
+        } else {
+            return isServerSide() ? 0 : nextJssInstanceId++;
+        }
+    });
```
Comment (atabel) [inline src/theme-context-provider.tsx:90]: do you need the useState? why don't you do this logic inside the useMemo?

**Review by atabel — APPROVED** [review]

File: src/theme-context-provider.tsx (hunk 2 above)
Comment (pladaria) [reply src/theme-context-provider.tsx:90]: I don't want the useMemo to depend on the prop. Once the value is set I don't want to allow it to change. A better option would be to use a ref but useRef doesn't accept a callback

**Pull request merged**